code/game/objects/items/weapons/storage/lockbox.dm

```diff
@@ -107,51 +118,67 @@
 	icon_broken = "medalbox+b"
 	can_hold = list(/obj/item/clothing/accessory/medal)
 
+/obj/item/weapon/storage/lockbox/medal/AltClick()
+	if(!locked)
+		open = (open ? FALSE : TRUE)
+		update_icon()
+	..()
+
 /obj/item/weapon/storage/lockbox/medal/PopulateContents()
+	new /obj/item/clothing/accessory/medal/gold/captain(src)
+	new /obj/item/clothing/accessory/medal/silver/valor(src)
 	new /obj/item/clothing/accessory/medal/silver/valor(src)
+	new /obj/item/clothing/accessory/medal/silver/security(src)
 	new /obj/item/clothing/accessory/medal/bronze_heart(src)
+	new /obj/item/clothing/accessory/medal/plasma/nobel_science(src)
+	new /obj/item/clothing/accessory/medal/plasma/nobel_science(src)
 	for(var/i in 1 to 3)
 		new /obj/item/clothing/accessory/medal/conduct(src)
-	new /obj/item/clothing/accessory/medal/gold/captain(src)
-	new /obj/item/clothing/accessory/medal/silver/security(src)
-	new /obj/item/clothing/accessory/medal/plasma(src)
-	new /obj/item/clothing/accessory/medal/plasma/nobel_science(src)
-	new /obj/item/clothing/accessory/medal/gold/heroism(src)
 
-/obj/item/weapon/storage/lockbox/secmedal
+/obj/item/weapon/storage/lockbox/medal/update_icon()
+	cut_overlays()
+	if(locked)
+		icon_state = "medalbox+l"
+		open = FALSE
+	else
+		icon_state = "medalbox"
+		if(open)
+			icon_state += "open"
+		if(broken)
+			icon_state += "+b"
+		if(contents && open)
+			for (var/i in 1 to contents.len)
+				var/medaltype = "medal"
+				var/obj/item/clothing/accessory/medal/M = contents[i]
+				if(istype(M, /obj/item/clothing/accessory/medal/gold))

```

Comment:
Remie doesn't like to actually explain why code is bad, just point out that it is.

In this case, the issue is the `istype` checks, which are generally viewed as hacky code. Instead of doing that, I'd recommend you add the `medaltype` variable to the base medal object, then when you're assigning the overlays to the medal box, use that instead of istype checks. With that, you could remove the `medaltype` define on line 151, as well as cutting out all the `istype` checks, and replacing the icon define with `var/icon/medalicon = icon(initial(icon), M.medaltype)` so it pulls from the medal.

You might also want to check if the medal actually has that variable defined and, if not, use a generic one like "-silver" or something. Even if you have sprites for all the medals right now, it's good to cover any future cases.